hilti/toolchain/src/compiler/codegen/codegen.cc

```diff
@@ -554,6 +554,16 @@ hilti::Unit* CodeGen::hiltiUnit() const {
     return _hilti_unit;
 }
 
+codegen::TypeUsage CodeGen::parameterKindToTypeUsage(declaration::parameter::Kind k) {
+    switch ( k ) {
+        case declaration::parameter::Kind::Copy: return codegen::TypeUsage::CopyParameter;
+        case declaration::parameter::Kind::In: return codegen::TypeUsage::InParameter;
+        case declaration::parameter::Kind::InOut: return codegen::TypeUsage::InOutParameter;
+        case declaration::parameter::Kind::Unknown: logger().internalError("parameter kind not set");
+        default: util::cannot_be_reached();
```

Comment:
Do we need this here? I would much prefer to reduce the number of `default` cases as much as possible as during updates to the enum they can lead to silently compiling but misbehaving code.

Reply:
I'll move it to behind the switch. The problem is that GCC has trouble recognizing that all enum values have been covered and will warn about not returning something.